modules/ROOT/pages/kernel-args.adoc

```diff
@@ -1,10 +1,87 @@
+You can append kernel argument. This is useful with e.g. `console=` that can be used multiple times. An empty value for an argument is allowed:
+
+[source,bash]
+----
+$ sudo rpm-ostree kargs --append=KEY=VALUE
+----
+
+.Exemple: Add reserved memory for Kdump support
+
+[source,bash]
+----
+$ sudo rpm-ostree kargs --append='crashkernel=256M'
+----
+
+See also xref:debugging-kernel-crashes.adoc[Debugging kernel crashes using kdump].
+
+== Removing existing kernel arguments
+
+You can delete a specific kernel argument key/value pair or an entire argument with a single key/value pair:
+
+[source,bash]
+----
+$ sudo rpm-ostree kargs --delete=KEY=VALUE
+----
+
+.Exemple: Remove console parameters to enable kernel auto-detection
+
+[source,bash]
+----
+$ sudo rpm-ostree kargs --delete 'console=ttyS0,115200n8'
+----
+
+See also xref:emergency-shell.adoc[Emergency console access].
+
+.Exemple: Update an existing system from cgroupsv1 to cgroupsv2 and immediately reboot
+
+[source,bash]
+----
+$ sudo rpm-ostree kargs --delete=systemd.unified_cgroup_hierarchy --reboot
+----
+
+== Replacing existing kernel arguments
+
+You can replace an existing kernel argument with a new value. You can direclty use `KEY=VALUE` if only one value exist for that argument:
+[source,bash]
+----
+$ sudo rpm-ostree kargs --replace=KEY=VALUE=NEWVALUE
+----
+
+.Exemple: Disable all CPU vulnerability mitigations
+
+[source,bash]
+----
+$ sudo rpm-ostree kargs --replace=mitigations=auto,nosmt=off
+----
+
+This switches `mitigations=auto,nosmt` to `mitigations=off` to disable all CPU vulnerability mitigations.
+
+== Interactive editing
+
+To use an editor to modify the kernel arguments:
+
+[source,bash]
+----
+$ sudo rpm-ostree kargs --editor
+----
+
+== Modifying Kernel Arguments via Ignition
+
+Currently to change kernel arguments via Ignition, you must script a systemd service which runs `rpm-ostree kargs` and then trigger a reboot.
+
+NOTE: In the future, we will have a more Ignition-friendly method of doing this with stronger guarantees. See upstream issues https://github.com/coreos/ignition/issues/1051[ignition#1051], https://github.com/coreos/fedora-coreos-tracker/issues/318[fedora-coreos-tracker#318] and https://github.com/coreos/fedora-coreos-tracker/issues/752[fedora-coreos-tracker#752] for more information.
+
+NOTE:  The `After=systemd-machine-id-commit.service` directive is important in the following examples to avoid some subtle issues. Similarly, any `ConditionFirstBoot=true` services should use `Before=first-boot-complete.target systemd-machine-id-commit.service`. See https://github.com/systemd/systemd/blob/3045c416e1cbbd8ab40577790522217fd1b9cb3b/man/systemd.unit.xml#L1315[the systemd documentation] for more details.
+
+=== Example: Staying on cgroups v1
+
+Cgroups v2 is now the default on Fedora CoreOS. Here's an example which adds the `systemd.unified_cgroup_hierarchy=0` kernel argument so that the machine keeps using cgroups v1:
```

Comment:
```suggestion
Cgroups v2 is now the default on Fedora CoreOS on the `next` stream. Here's an example which adds the `systemd.unified_cgroup_hierarchy=0` kernel argument so that the machine keeps using cgroups v1:
```